src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJobInd.java

```diff
@@ -0,0 +1,106 @@
+    private UserCommonsRepository userCommonsRepository;
+    @Getter
+    private UserRepository userRepository;
+    @Getter
+    private ProfitRepository profitRepository;
+    @Getter
+    private long commonsID;
+
+    public String formatDollars(double amount) {
+        return  String.format("$%.2f", amount);
+    }
+
+    @Override
+    public void accept(JobContext ctx) throws Exception {
+        ctx.log("Starting to milk the cows");
+        Optional<Commons> commonMilkedOpt = commonsRepository.findById(commonsID);
+
+        if(commonMilkedOpt.isPresent()){
+            Commons commonMilked = commonMilkedOpt.get();
+            String name = commonMilked.getName();
+            double milkPrice = commonMilked.getMilkPrice();
+            ctx.log("Milking cows for Commons: " + name + ", Milk Price: " + formatDollars(milkPrice));
+
+            Iterable<UserCommons> allUserCommons = userCommonsRepository.findByCommonsId(commonMilked.getId());
+
+            for (UserCommons userCommons : allUserCommons) {
+                milkCows(ctx, commonMilked, userCommons);
+            }
+            
+
+            ctx.log("Cows have been milked!");
+        } else {
+            ctx.log(String.format("No commons found for id %d", commonsID));
+        }
+    }
+
+    /** This method performs the function of milking the cows for a single userCommons.
+     *  It is a public method only so it can be exposed to the unit tests
+     * @param ctx the JobContext
+     * @param commons the Commons
+     * @param userCommons the UserCommons
+     *
+     */
+
+    public void milkCows(JobContext ctx, Commons commons, UserCommons userCommons) {
+        User user = userCommons.getUser();
+
+        ctx.log("User: " + user.getFullName()
+                + ", numCows: " + userCommons.getNumOfCows()
+                + ", cowHealth: " + userCommons.getCowHealth()
+                + ", totalWealth: " + formatDollars(userCommons.getTotalWealth()));
+
+        double profitAmount = calculateMilkingProfit(commons, userCommons);
+        Profit profit = Profit.builder()
+                .userCommons(userCommons)
+                .amount(profitAmount)
+                .timestamp(LocalDateTime.now())
+                .numCows(userCommons.getNumOfCows())
+                .avgCowHealth(userCommons.getCowHealth())
+                .build();
+        double newWeath = userCommons.getTotalWealth() + profitAmount;
+        userCommons.setTotalWealth(newWeath);
+        userCommonsRepository.save(userCommons);
+        profit = profitRepository.save(profit);
+        ctx.log("Profit for user: " + user.getFullName()
+                + " is: " + formatDollars(profitAmount)
+                + ", newWealth: " + formatDollars(newWeath));
+    }
+
+    /**
+     * Calculate the profit for a user from milking their cows.
+     *
+     * @param userCommons
+     * @return
+     */
+    public static double calculateMilkingProfit(Commons commons, UserCommons userCommons) {
```

Comment:
For this one, there's definitely no reason to make a new method, when there is an existing public static method at the link below in `MilkTheCows` that's identical.  Just call that (or move that method when you move the code for milkCows).

https://github.com/ucsb-cs156-m23/proj-happycows-m23-10am-4/blob/6231df7533e393a376dbd4edc4b4a06eeeb4592a/src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java#L94